src/core/StelTexture.cpp

```diff
@@ -281,12 +279,21 @@ bool StelTexture::getDimensions(int &awidth, int &aheight)
 	return true;
 }
 
-QByteArray StelTexture::convertToGLFormat(const QImage& image, GLint& format, GLint& type)
+QByteArray StelTexture::convertToGLFormat(QImage image, GLint& format, GLint& type, int& width, int& height)
 {
 	QByteArray ret;
-	const int width = image.width();
-	const int height = image.height();
-	if (StelMainView::getInstance().getGLInformation().supportsLuminanceTextures && image.isGrayscale())
+	const auto glInfo = StelMainView::getInstance().getGLInformation();
+	width = std::min(image.width(), glInfo.maxTextureSize);
+	height = std::min(image.height(), glInfo.maxTextureSize);
+	if(width != image.width() || height != image.height())
+	{
+		qWarning().nospace() << "Got a texture with too large dimensions: "
+							 << image.width() << "x" << image.height()
+							 << ", while maximum size is " << glInfo.maxTextureSize;
```

Comment:
The warning leaves the user in doubt what happens: is there a failure now?
Better wording: 
```
qWarning().nospace() << "Texture too large: "
							 << image.width() << "x" << image.height()
							 << ", shrinking to allowed maximum of" << glInfo.maxTextureSize;
```

Reply:
Updated.